core/src/main/scala/akka/persistence/r2dbc/internal/R2dbcExecutor.scala

```diff
@@ -57,6 +57,38 @@ import reactor.core.publisher.Flux
       promise.future
     }
 
+    def asFutureSeq(): Future[immutable.IndexedSeq[T]] =
+      asFutureSeq(identity)
+
+    def asFutureSeq[U](mapFunc: T => U): Future[immutable.IndexedSeq[U]] = {
```

Comment:
This is becoming too much "implement your own reactive streams ...". The reason for doing that for the simple Future and Future[Done] cases are that those are used for very fine grained operations and I want them to be as fast as possible.

Here we should use the reactor api or Akka Streams `Source.fromPublisher`.

Reply:
Probably best to use the reactor api since we already use that in updateBatchInTx

Reply:
I didn't want to use Source.fromPublisher because we had issues with iterations with the driver. Although I think here is different. The issue we had was in updating. 

I did a first impl using reactor API, but changed to this. The reason I changed is that I was planning to use the same in the updateBatchInTx. I want to have one single way of taking a Flux and building a Scala IndexedSeq out of it without repeating what we did in updateBatchInTx.

Anyway, I think we add a method to a Flux that does that without using subscription. I will try this out.